source/material_model/interface.cc

```diff
@@ -393,6 +399,85 @@ namespace aspect
       reaction_terms(n_points, std::vector<double>(n_comp, numbers::signaling_nan<double>()))
     {}
 
+    namespace internal
+    {
+      template <int dim>
+      void check(const std::vector<double> &quantity,
+                 const char *name,
+                 const unsigned int n_points)
+      {
+        Assert(quantity.size() == n_points,
+               ExcMessage("Quantity '"+std::string(name)+"' has an incorrect size!"));
+        for (unsigned int i=0; i<n_points; ++i)
+          Assert(!std::isnan(quantity[i]),
```

Comment:
This and the following 4 versions of the function only differ in how the check `isnan(...)`. I think it would easier if you just overload the `isnan` functions and make the `check` function generic for type `T`.

deal.II has `numbers::is_nan()`, but that exists only for `double`. Maybe extend that in the ASPECT utilities namespace to other data types?